L1Trigger/CSCTriggerPrimitives/src/CSCRPCMotherboard.cc

```diff
@@ -0,0 +1,728 @@
+    const int strip_bx = getBX(p.second)+lct_central_bx;
+    if (std::abs(clct.getBX()-strip_bx)>1) continue;
+    if (std::abs(lowStrip - stripRoll) <= maxDeltaStrip_ or std::abs(stripRoll - highStrip) <= maxDeltaStrip_){
+      result.push_back(p);
+    }
+  }
+}
+
+
+void
+CSCRPCMotherboard::matchingRPCDigis(const CSCALCTDigi& alct, RPCDigiIds& result) const
+{
+  // no matching digis for invalid stub
+  if (not alct.isValid()) return;
+
+  const int alctRoll = (getLUT()->get_csc_wg_to_rpc_roll(par))[alct.getKeyWG()];
+  RPCDigiIdsBX lut = rpcDigis_;
+  for (const auto& p: lut[alct.getBX()]){
+    auto stripRoll(getRoll(p));
+    if (alctRoll !=  stripRoll) continue;
+    result.push_back(p);
+  }
+}
+
+
+void
+CSCRPCMotherboard::matchingRPCDigis(const CSCCLCTDigi& clct,
+				    const CSCALCTDigi& alct,
+				    RPCDigiIds& result) const
+{
+  RPCDigiIds digisClct, digisAlct;
+
+  // digis matching to the CLCT
+  matchingRPCDigis(clct, digisClct);
+
+  // digis matching to the ALCT
+  matchingRPCDigis(alct, digisAlct);
+
+  // collect all *common* digis
+  intersection(digisClct, digisAlct, result);
+}
+
+void
+CSCRPCMotherboard::matchingRPCDigis(const CSCCLCTDigi& clct1,
+				    const CSCCLCTDigi& clct2,
+				    const CSCALCTDigi& alct1,
+				    const CSCALCTDigi& alct2,
+				    RPCDigiIds& result) const
+{
+  RPCDigiIds digisClct, digisAlct;
+
+  // digis matching to the CLCTs
+  matchingRPCDigis(clct1, clct2, digisClct);
+
+  // digis matching to the ALCTs
+  matchingRPCDigis(alct1, alct2, digisAlct);
+
+  // collect *all* matching digis
+  result.reserve(digisClct.size() + digisAlct.size());
+  result.insert(std::end(result), std::begin(digisClct), std::end(digisClct));
+  result.insert(std::end(result), std::begin(digisAlct), std::end(digisAlct));
+}
+
+unsigned int CSCRPCMotherboard::findQualityRPC(const CSCALCTDigi& aLCT, const CSCCLCTDigi& cLCT, bool hasRPC) const
+{
+  unsigned int quality = 0;
+  /*
+    Same LCT quality definition as standard LCTs
+    a4 and c4 takes RPCs into account!!!
+  */
+  // 2008 definition.
+  if (!(aLCT.isValid()) || !(cLCT.isValid())) {
+    if (aLCT.isValid() && !(cLCT.isValid()))      quality = 1; // no CLCT
+    else if (!(aLCT.isValid()) && cLCT.isValid()) quality = 2; // no ALCT
+    else quality = 0; // both absent; should never happen.
+  }
+  else {
+    const int pattern(cLCT.getPattern());
+    if (pattern == 1) quality = 3; // layer-trigger in CLCT
+    else {
+      // ALCT quality is the number of layers hit minus 3.
+      // CLCT quality is the number of layers hit.
+      int n_rpc = 0;
+      if (hasRPC) n_rpc = 1;
+      const bool a4((aLCT.getQuality() >= 1 and aLCT.getQuality() != 4) or
+		    (aLCT.getQuality() == 4 and n_rpc >=1));
+      const bool c4((cLCT.getQuality() >= 4) or (cLCT.getQuality() >= 3 and n_rpc>=1));
+      //              quality = 4; "reserved for low-quality muons in future"
+      if      (!a4 && !c4) quality = 5; // marginal anode and cathode
```

Comment:
Define enum(s) for these quality values

Reply:
I don't see the point of doing that. Such enum would not be used elsewhere outside that function.

Reply:
`findQualityRPC()` returns the quality value, so it must be used elsewhere.

In 10 years from now, will anyone know what these magic numbers mean?